app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -472,13 +498,50 @@ class PipelineSampleReport extends React.Component {
     } else {
       excluded_categories.push(e.target.value);
     }
+    // Also update subcategory to match category
+    let new_exclude_subcats = this.state.exclude_subcats;
+    let subcats = Object.keys(this.category_child_parent);
+    for (let i = 0; i < subcats.length; i++) {
+      let subcat = subcats[i];
+      let parent = this.category_child_parent[subcat]
+      let i_excluded_categories = excluded_categories.indexOf(parent)
```

Comment:
I will rewrite this whole logic as: 

let parent_excluded = (excluded_categories.indexOf(parent) >= 0); 

if (parent_excluded) {
    //subcat should be excluded 
   if (new_exclude_subcats.indexOf(subcat) <= 0) {
      new_exclude_subcats.push(subcat);
  }
} else {
  let idx = new_exclude_subcats.indexOf(subcat); 
  if (idx >= 0) {
    new_exclude_subcats.splice(idx, 1);
}
}